cluster-autoscaler/cloudprovider/azure/azure_cloud_provider.go

```diff
@@ -122,8 +122,12 @@ func (azure *AzureCloudProvider) NodeGroupForNode(node *apiv1.Node) (cloudprovid
 }
 
 // HasInstance returns whether a given node has a corresponding instance in this cloud provider
-func (azure *AzureCloudProvider) HasInstance(*apiv1.Node) (bool, error) {
-	return true, cloudprovider.ErrNotImplemented
+func (azure *AzureCloudProvider) HasInstance(node *apiv1.Node) (bool, error) {
+	ng, err := azure.azureManager.azureCache.FindForInstance(&azureRef{Name: node.Spec.ProviderID}, azure.azureManager.azureCache.vmType)
```

Comment:
This can be reduced to
```suggestion
	ng, err := azure.azureManager.GetNodeGroupForInstance(&azureRef{Name: node.Spec.ProviderID})
```
or even further to just
```suggestion
	ng, err := azure.NodeGroupForNode(node)
```

But this brings up several questions. What exactly are the semantics of `HasInstance` and `NodeGroupForNode`? If `HasInstance` can be correctly and reliably implemented by `NodeGroupForNode` (or is it a special property of our provider?), why would `HasInstance` be needed in the CP interface to begin with? `NodeGroupForNode` returning nil is supposed to mean "instance should not be handled by autoscaler", not that it does not exist. Using it (or equivalent logic) for implementing `HasInstance` would mean NG node will always be reported as existing, non-NG node will always be reported as deleted; which seems wrong, probably need to think this through.